module-code/app/securesocial/core/providers/ChatworkProvider.scala

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2015 Mikael Vallerie
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package securesocial.core.providers
+
+import play.api.libs.json.{ Reads, Json, JsValue }
+import play.api.{ Configuration, Environment }
+import play.api.libs.ws.{ WSAuthScheme, WSResponse, WSClient }
+import securesocial.core
+import securesocial.core._
+import securesocial.core.services.{ CacheService, HttpService, RoutesService }
+
+import scala.concurrent.{ ExecutionContext, Future }
+import ChatworkProvider.{ ErrorResponse, UserResponse }
+import ChatworkProvider._
+
+class ChatworkOAuth2Client(
+    httpService: HttpService, settings: OAuth2Settings)(implicit executionContext: ExecutionContext) extends OAuth2Client.Default(httpService, settings)(executionContext) {
+  override def exchangeCodeForToken(code: String, callBackUrl: String, builder: OAuth2InfoBuilder): Future[OAuth2Info] = {
+    val params = Map(
+      OAuth2Constants.GrantType -> Seq(OAuth2Constants.AuthorizationCode),
+      OAuth2Constants.Code -> Seq(code),
+      OAuth2Constants.RedirectUri -> Seq(callBackUrl)
+    ) ++ settings.accessTokenUrlParams.mapValues(Seq(_))
+    httpService.url(settings.accessTokenUrl)
+      .withAuth(settings.clientId, settings.clientSecret, WSAuthScheme.BASIC)
+      .post(params).map(builder)
+  }
+
+}
+/**
+ * A Chatwork provider
+ *
+ */
+class ChatworkProvider(routesService: RoutesService,
+  cacheService: CacheService,
+  client: OAuth2Client)(implicit val configuration: Configuration, val playEnv: Environment)
+    extends OAuth2Provider.Base(routesService, client, cacheService) {
+  override val id = ChatworkProvider.Chatwork
+  private val Logger = play.api.Logger("securesocial.core.providers.ChatworkProvider")
```

Comment:
`OAuth2Provider` has a member named `logger`, so we don't need this.

Reply:
I replace original logger to inherited logger.